office365_connector.py

```diff
@@ -486,30 +424,108 @@ def load_state(self):
-            self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR, str(e)))
-            state = None
-
-        return state
+        if not isinstance(state, dict):
+            self.debug_print("Resetting the state file with the default format")
+            state = {
+                "app_version": self.get_app_json().get('app_version')
+            }
+            return state
+        return self._decrypt_state(state, self._asset_id)
 
     def save_state(self, state):
         """
-        Encrypt and save the current state dictionary to the the state file.
+        Encrypt and save the current state dictionary to the state file.
 
         :param state: state dictionary
         :return: status
         """
-        try:
-            state = _encrypt_state(state, self._asset_id)
-        except Exception as e:
-            self._dump_error_log(e)
-            self.debug_print("{}: {}".format(MSGOFFICE365_ENCRYPTION_ERROR, str(e)))
-            return phantom.APP_ERROR
+        return super().save_state(self._encrypt_state(state, self._asset_id))
+
+    def _decrypt_state(self, state, salt):
+        """
+        Decrypts the state.
+
+        :param state: state dictionary
+        :param salt: salt used for decryption
+        :return: decrypted state
+        """
+        if not state.get("is_encrypted"):
+            return state
+
+        if state.get("non_admin_auth", {}).get("access_token"):
+            try:
+                state["non_admin_auth"]["access_token"] = encryption_helper.decrypt(
+                    state["non_admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["non_admin_auth"]["access_token"] = None
+
+        if state.get("non_admin_auth", {}).get("refresh_token"):
+            try:
+                state["non_admin_auth"]["refresh_token"] = encryption_helper.decrypt(
+                    state["non_admin_auth"]["refresh_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["non_admin_auth"]["refresh_token"] = None
+
+        if state.get("admin_auth", {}).get("access_token"):
+            try:
+                state["admin_auth"]["access_token"] = encryption_helper.decrypt(
+                    state["admin_auth"]["access_token"], salt
+                )
+            except Exception as ex:
+                self.debug_print("{}: {}".format(MSGOFFICE365_DECRYPTION_ERROR,
+                                                              _get_error_msg_from_exception(ex, self)))
+                state["admin_auth"]["access_token"] = None
+
```

Comment:
Shouldn't this set `state["is_encrypted"] = False`?

Reply:
Programmatically, It isn't required, because it won't be checked, the reason for adding the 'is_encrypted' key in the state file is to handle backward compatibility. If we upgrade the app from any version to another version that has 'is_encrypted' added, we should check this condition before decrypting tokens that have already been decrypted.